tests/integration_tests/functional/test_cpu_template_helper.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests that verify the cpu-template-helper's behavior."""
+
+import json
+import platform
+from pathlib import Path
+
+import pytest
+
+from framework import defs, utils
+from host_tools import cargo_build
+
+PLATFORM = platform.machine()
+
+
+# pylint: disable=too-few-public-methods
+class CpuTemplateHelper:
+    """
+    Class for CPU template helper tool.
+    """
+
+    # Class constants
+    BINARY_NAME = "cpu-template-helper"
+    BINARY_PATH = Path(
+        f"{defs.FC_WORKSPACE_TARGET_DIR}/"
+        f"{cargo_build.DEFAULT_BUILD_TARGET}/"
+        f"release/{BINARY_NAME}"
+    )
+
+    def __init__(self):
+        """Build CPU template helper tool binary"""
+        if not self.BINARY_PATH.exists():
+            utils.run_cmd(f"cd {defs.FC_WORKSPACE_DIR}")
```

Comment:
I think you can pass `cwd` parameter to run_cmd to do this. Otherwise I don't think the `cd` will be effective in the parent process.